bin/generate_truss_examples.py

```diff
@@ -0,0 +1,301 @@
+        print(f"Error cloning the repo: {e}")
+        sys.exit(1)
+
+
+def fetch_file_contents(path: str):
+    with open(path, "r") as f:
+        return f.read()
+
+
+def _fetch_example_dirs(root_dir: str) -> List[str]:
+    """
+    Walk through the directory structure from the root directory and
+    find all directories that have the specified file in it.
+    """
+    dirs_with_file = []
+
+    for dirpath, _, filenames in os.walk(root_dir):
+        if DOC_CONFIGURATION_FILE in filenames:
+            dirs_with_file.append(dirpath)
+
+    return dirs_with_file
+
+
+def _get_example_destination(truss_directory: str) -> Path:
+    """
+    Get the destination directory for the example.
+    """
+    original_path = Path(truss_directory)
+    folder, example = original_path.parts[1:]
+    example_file = f"{example}.mdx"
+    return Path("docs/examples") / folder / example_file
+
+
+def _get_file_type(file_path: str) -> FileType:
+    _, extension = os.path.splitext(file_path)
+    if extension == ".yaml":
+        return FileType.YAML
+
+    if extension == ".py":
+        return FileType.PYTHON
+
+    raise ValueError(f"Unknown file type: {extension}")
+
+
+class ContentBlock:
+    def formatted_content(self) -> str:
+        raise NotImplementedError
+
+
+class CodeBlock(ContentBlock):
+    def __init__(self, file_type: FileType, file_path: str):
+        self.file_type = file_type
+        self.file_path = file_path
+        self.content = ""
+
+    def formatted_content(self) -> str:
+        """
+        Outputs code blocks in the format:
+
+        ```python main.py
+        def main():
+            ...
+        ```
+        """
+        return f"\n```{self.file_type.value} {self.file_path}\n{self.content}```"
+
+
+class MarkdownBlock(ContentBlock):
+    def __init__(self, content: str):
+        self.content = content
+
+    def formatted_content(self) -> str:
+        # Remove the first comment and space character, such that
+        # "# Hello" becomes "Hello
+        return self.content.strip()[2:]
+
+
+class MarkdownExtractor:
+    """
+    Class that supports ingesting a code file line-by-line, and produces a formatted
+    mdx file.
+    """
+
+    def __init__(self, file_type: FileType, file_path: str):
+        self.file_type = file_type
+        self.file_path = file_path
+
+        self.blocks: List[ContentBlock] = []
+        self.current_code_block: Optional[CodeBlock] = None
+
+    def ingest(self, line: str):
+        """
+        For each line, check that it is a comment by the presence of "#".
+        If it is a comment, append it to the blocks.
+
+        If it is not a comment, either append to the current code block, or
+        create a new code block if this isn't one.
+
+        When this is finished, we can then very easily produce the mdx file.
+        """
+        stripped_line = line.strip()
+
+        # Case of Markdown line
+        if stripped_line.startswith("#"):
```

Comment:
Do we need to handle multiple comments/docstrings?

Reply:
I thought about this -- parsing out docstrings is a little tough. Since for the examples that we are including in the docs we are in charge of, I thin it's easier if we just choose to use the "#" style comments to write the "doc" part of this